docs/zksync-era-integration.md

```diff
@@ -0,0 +1,184 @@
+
+At the most basic level, the Bootloader performs the following steps:
+
+1. Reads the initial batch information and makes a call to the SystemContext contract to validate the batch.
+2. Loops through all transactions and executes them until the `execute` flag is set to $0$, at that point, it jumps to step 3. <!-- Here we could also add that the transaction gets processed based on where it came from (l1 or l2) -->
+3. Seals l2 block and publish final data to the l1.
+
+The initial validation of the batch is necessary, since, as we'll see below, the Bootloader starts with its memory pre-filled with any data the operator wants. That is why it needs to validate its correctness.
+
+For more details, you can see the [main loop](https://github.com/matter-labs/era-contracts/blob/main/system-contracts/Bootloader/bootloader.yul#L3962-L3965) or the [full contract code](https://github.com/matter-labs/era-contracts/blob/main/system-contracts/bootloader/bootloader.yu).
+
+## Operator/sequencer
+
+Currently, the operator is a centralized server (there are plans to make a decentralized consensus for operators) which can be thought of as the node entry point, its responsibilities are:
+
+-   Initializing the `era_vm` and keeping its state.
+-   Orchestrating the Bootloader and keeping its state.
+-   Keeping storage database and committing changes.
+
+Here is a simplified version of what the vm on the operator end looks like:
+
+```rust
+struct OperatorVm {
+    pub(crate) inner: EraVM, // this would be the actual `era_vm`
+    pub suspended_at: u16, // last pc when execution stopped because of a hook
+    pub Bootloader_state: BootloaderState,
+    pub(crate) storage: StorageDb,
+    pub snapshot: Option<VmSnapshot>,
+}
+```
+
+### Initializing the vm
+
+VM initialization involves:
+
+-   Loading the Bootloader bytecode and initializing its state.
+-   Setting up the `era_vm` by injecting the Bootloader code, loading default contracts, and configuring other settings.
+
+When setting up `era_vm`, the operator provides access to the chain storage database with the following API:
+
+```rust
+trait Storage {
+    fn decommit(&mut self, hash: U256) -> Option<Vec<U256>>;
+
+    fn storage_read(&mut self, key: &StorageKey) -> Option<U256>;
+
+    fn cost_of_writing_storage(&mut self, key: &StorageKey, value: U256) -> u32;
+
+    fn is_free_storage_slot(&self, key: &StorageKey) -> bool;
+}
+```
+
+This storage is saved in the VM state as a pointer. Here’s a brief explanation of each function:
+
+-   **decommit**: given a hash it returns a contract bytecode from the database.
```

Comment:
> given a hash it returns a contract bytecode from the database.

given a hash it returns *its corresponding bytecode* (if it exists) from the database